commands.go

```diff
@@ -84,26 +93,38 @@ func cmdMigrationsUp(c *cli.Context) error {
 		reportError(mtx.stateProvider.Config(), "up until stop", errors.New("No such migration file: "+stopAfter))
 		return errAbort
 	}
+	prettyWidth := largestWithOf(all)
 	for _, each := range all {
 		log.Println(statusSeparator)
-		log.Println(execDo, pretty(each.Filename))
-		if err := ExecuteAll(each.DoSection, mtx.config().shellEnv(), c.GlobalBool("v")); err != nil {
-			reportError(mtx.stateProvider.Config(), "do", err)
-			return errAbort
+		leadingTitle := execDo
+		if isLogOnly {
+			leadingTitle = execPlan
 		}
-		mtx.lastApplied = each.Filename
-		// save after each succesful migration
-		if err := mtx.stateProvider.SaveState(mtx.lastApplied); err != nil {
-			reportError(mtx.stateProvider.Config(), "save state", err)
-			return errAbort
+		log.Printf("%s %-"+strconv.Itoa(prettyWidth)+"s (%s)\n", leadingTitle, pretty(each.Filename), each.Filename)
+		if isLogOnly {
```

Comment:
Extract to a func that accepts 1 param (isLogOnly) 

will reduce the size of this function significantly maybe a runAll private function?